tasks/src/resolver.rs

```diff
@@ -0,0 +1,87 @@
+use imap_flow::{client::ClientFlow, Flow, FlowInterrupt};
+use imap_types::response::{Response, Status};
+use tracing::warn;
+
+use crate::{Scheduler, SchedulerError, SchedulerEvent, Task, TaskHandle};
+
+/// The resolver is a scheduler than manages one task at a time.
+pub struct Resolver {
+    scheduler: Scheduler,
+}
+
+impl Flow for Resolver {
+    type Event = SchedulerEvent;
+    type Error = SchedulerError;
+
+    fn enqueue_input(&mut self, bytes: &[u8]) {
+        self.scheduler.enqueue_input(bytes);
+    }
+
+    fn progress(&mut self) -> Result<Self::Event, FlowInterrupt<Self::Error>> {
+        self.scheduler.progress()
+    }
+}
+
+impl Resolver {
+    /// Create a new resolver.
+    pub fn new(flow: ClientFlow) -> Self {
```

Comment:
I like that the `Scheduler` is just an implementation of `Resolver` and does not appear in its API.

Reply:
That said, the resolver still uses scheduler events and errors, is it OK? Should type aliases or struct/enum wrappers be defined?

Reply:
It's okay. There are different alternatives how we could proceed and I don't want to decide it now. The solution should probably also include the unexpected cases that we log at the moment.